### PR TITLE
feat: improve numaflow k8s events

### PR DESCRIPTION
**Pull request by dpadhiar**
Explain what this PR does.

Currently Numaflow is emitting events for the pipeline and vertex controllers. This PR adds events for creating updating and deleting resources for the pipeline, vertex and now isbsvcs.

### CODE REVIEW

File: pkg/reconciler/isbsvc/installer/jetstream.go [hunk 1]
```diff
@@ -104,21 +107,25 @@ func (r *jetStreamInstaller) Install(ctx context.Context) (*dfv1.BufferServiceCo
 	if err := r.createSecrets(ctx); err != nil {
 		r.logger.Errorw("Failed to create jetstream auth secrets", zap.Error(err))
 		r.isbSvc.Status.MarkDeployFailed("JetStreamAuthSecretsFailed", err.Error())
+		r.recorder.Eventf(r.isbSvc, corev1.EventTypeNormal, "JetStreamAuthSecretsFailed", "Failed to create jetstream auth secrets: %v", err.Error())
```
Comment (whynowy) [inline pkg/reconciler/isbsvc/installer/jetstream.go:110]: Shouldn't be Normal.

File: pkg/reconciler/isbsvc/installer/jetstream.go [hunk 2]
```diff
@@ -104,21 +107,25 @@ func (r *jetStreamInstaller) Install(ctx context.Context) (*dfv1.BufferServiceCo
...
 		return nil, err
 	}
 	if err := r.createConfigMap(ctx); err != nil {
 		r.logger.Errorw("Failed to create jetstream ConfigMap", zap.Error(err))
 		r.isbSvc.Status.MarkDeployFailed("JetStreamConfigMapFailed", err.Error())
+		r.recorder.Eventf(r.isbSvc, corev1.EventTypeNormal, "JetStreamConfigMapFailed", "Failed to create jetstream ConfigMap: %v", err.Error())
```
Comment (whynowy) [inline pkg/reconciler/isbsvc/installer/jetstream.go:116]: Shouldn't be Normal.

File: pkg/reconciler/isbsvc/installer/jetstream.go [hunk 3]
```diff
@@ -104,21 +107,25 @@ func (r *jetStreamInstaller) Install(ctx context.Context) (*dfv1.BufferServiceCo
...
 		return nil, err
 	}
 	if err := r.createService(ctx); err != nil {
 		r.logger.Errorw("Failed to create jetstream Service", zap.Error(err))
 		r.isbSvc.Status.MarkDeployFailed("JetStreamServiceFailed", err.Error())
+		r.recorder.Eventf(r.isbSvc, corev1.EventTypeNormal, "JetStreamServiceFailed", "Failed to create jetstream Service: %v", err.Error())
```
Comment (whynowy) [inline pkg/reconciler/isbsvc/installer/jetstream.go:122]: Shouldn't be Normal.

File: pkg/reconciler/isbsvc/installer/jetstream.go [hunk 4]
```diff
@@ -104,21 +107,25 @@ func (r *jetStreamInstaller) Install(ctx context.Context) (*dfv1.BufferServiceCo
...
 		return nil, err
 	}
 	if err := r.createStatefulSet(ctx); err != nil {
 		r.logger.Errorw("Failed to create jetstream StatefulSet", zap.Error(err))
 		r.isbSvc.Status.MarkDeployFailed("JetStreamStatefulSetFailed", err.Error())
+		r.recorder.Eventf(r.isbSvc, corev1.EventTypeNormal, "JetStreamStatefulSetFailed", "Failed to create jetstream StatefulSet: %v", err.Error())
```
Comment (whynowy) [inline pkg/reconciler/isbsvc/installer/jetstream.go:128]: Shouldn't be Normal.

File: pkg/reconciler/pipeline/controller.go [hunk 5]
```diff
@@ -106,7 +106,6 @@ func (r *pipelineReconciler) reconcile(ctx context.Context, pl *dfv1.Pipeline) (
 				if err != nil {
 					logMsg := fmt.Sprintf("Failed to check if it's safe to delete the pipeline: %v", err.Error())
 					log.Error(logMsg)
```
Comment (whynowy) [inline pkg/reconciler/pipeline/controller.go:108]: Could you restructure the log above? At least put the pipeline name in it.

File: pkg/reconciler/vertex/controller.go [hunk 6]
```diff
@@ -105,14 +105,11 @@ func (r *vertexReconciler) reconcile(ctx context.Context, vertex *dfv1.Vertex) (
 		if apierrors.IsNotFound(err) {
 			e := fmt.Errorf("isbsvc %s not found", isbSvcName)
 			vertex.Status.MarkPhaseFailed("ISBSvcNotFound", e.Error())
-			r.recorder.Event(vertex, corev1.EventTypeWarning, "ISBSvcNotFound", e.Error())
 			return ctrl.Result{}, e
 		}
-		r.markPhaseLogEvent(vertex, log, "FindISBSvcFailed", err.Error(), "Failed to get ISB Service", zap.String("isbsvc", isbSvcName), zap.Error(err))
```
Comment (whynowy) [inline pkg/reconciler/vertex/controller.go:111]: Is this right? `markPhaseLogEvent` does 3 things, removing it means stopping doing all of them.

Comment (dpadhiar) [reply pkg/reconciler/vertex/controller.go:111]: This isn't right, should revert back to previously just marking the phase and logging the failure.

File: pkg/reconciler/vertex/controller.go [hunk 7]
```diff
@@ -105,14 +105,13 @@ func (r *vertexReconciler) reconcile(ctx context.Context, vertex *dfv1.Vertex) (
...
+		log.Errorw("Failed to get ISB Service", zap.String("isbsvc", isbSvcName), zap.Error(err))
+		vertex.Status.MarkPhaseFailed("FindISBSvcFailed", err.Error())
 		return ctrl.Result{}, err
 	}
 	if !isbSvc.Status.IsReady() {
-		r.markPhaseLogEvent(vertex, log, "ISBSvcNotReady", "isbsvc not ready", "isbsvc not ready", zap.String("isbsvc", isbSvcName))
```
Comment (whynowy) [inline pkg/reconciler/vertex/controller.go:115]: Please fix all the places like this.

**Review by whynowy — APPROVED** [review]

**Pull request merged**